src/main/java/org/spongepowered/common/mixin/core/block/MixinBlockState.java

```diff
@@ -103,9 +103,7 @@ public BlockSnapshot snapshotFor(Location<World> location) {
             .worldId(location.getExtent().getUniqueId());
         if (this.block.hasTileEntity() && location.getBlockType().equals(this.block)) {
             final TileEntity tileEntity = location.getTileEntity().get();
-            for (DataManipulator<?, ?> manipulator : tileEntity.getContainers()) {
-                builder.add(manipulator);
-            }
+            tileEntity.getContainers().forEach(builder::add);
```

Comment:
I think we should have something to automatically detect inner classes (including anonymous classes, lambdas and method references) in mixins and refuse to compile. Maybe this could make part of Mixin itself? It could even be done with javaparser.